### ISSUE TITLE
Don't trigger E402 on `pytest.importorskip` above imports

### ISSUE THREAD

**Issue by janosh** [astral-sh/ruff#13537]
Similar to ignoring `sys.path` (https://github.com/astral-sh/ruff/issues/10059) and `os.environ` manipulations above imports, I propose adding another exception for the widely used [`pytest.importorskip`](https://docs.pytest.org/en/stable/reference/reference.html#pytest-importorskip).

picking up on @ion-elgreco's https://github.com/astral-sh/ruff/issues/10059#issuecomment-2068900587, it may make sense to allow `warnings.filterwarnings` interspersed between imports as well.

example code snippet from https://github.com/materialsproject/atomate2/pull/993 where this came up repeatedly (would be nice to get rid off the many `# noqa: E402`):

```py
from atomate2.openff.utils import counts_from_box_size, merge_specs_by_name_and_smiles

pytest.importorskip("openff.toolkit")
import openff.toolkit as tk  # noqa: E402
from openff.interchange import Interchange  # noqa: E402
from openff.toolkit.topology import Topology  # noqa: E402
from openff.toolkit.topology.molecule import Molecule  # noqa: E402
```

**Comment by dhruvmanila** [comment]
For anyone interested, this requires adding a new function in https://github.com/astral-sh/ruff/blob/fdd0a22c03ddecea7e8b870b6d7a007210ee3d27/crates/ruff_python_semantic/src/analyze/imports.rs similar to `is_os_environ_modification` or `is_sys_path_modification` and updating the condition at https://github.com/astral-sh/ruff/blob/fdd0a22c03ddecea7e8b870b6d7a007210ee3d27/crates/ruff_linter/src/checkers/ast/mod.rs#L496-L502

Once https://github.com/astral-sh/ruff/issues/8794 is implemented, maybe we could only check for this in test files.

This should be added under preview mode.
